engine/gui/src/gui_script.cpp

```diff
@@ -2841,17 +2841,21 @@ namespace dmGui
      *
      * @name gui.is_enabled
      * @param node [type:node] node to query
+     * @param recursive [type:boolean] check hierarchy recursively
      * @return enabled [type:boolean] whether the node is enabled or not
      */
     static int LuaIsEnabled(lua_State* L)
     {
+        int top = lua_gettop(L);
         HNode hnode;
         InternalNode* n = LuaCheckNode(L, 1, &hnode);
         (void) n;
 
         Scene* scene = GuiScriptInstance_Check(L);
+        bool recursive = top == 2 && lua_toboolean(L, 2);
```

Comment:
What if I've accidentally called the function with three arguments?

`gui.pick_node(node, true, "foobar")`

Perhaps better to check `top >= 2`?

Reply:
Thank you, totally makes sense.